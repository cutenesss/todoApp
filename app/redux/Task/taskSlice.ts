import { createSlice, PayloadAction } from '@reduxjs/toolkit'
import { ItemTaskProps } from '../../models/TaskModels'

export interface ProductState {
  listTask: Array<ItemTaskProps>
}

const initialState: ProductState = {
  listTask: [],
}

interface editTaskBody {
  idOldItem: number
  newItem: ItemTaskProps
}

const TaskSlice = createSlice({
  name: 'task',
  initialState,
  reducers: {
    addTask(state, action: PayloadAction<ItemTaskProps>) {
      state.listTask = state.listTask.concat(action.payload)
    },
    editTask(state, action: PayloadAction<editTaskBody>) {
      const currentItemIndex = state.listTask.findIndex(
        task => task.id === action.payload.idOldItem
      )
      const newListTask = [...state.listTask]
      newListTask[currentItemIndex].description =
        action.payload.newItem.description
      newListTask[currentItemIndex].name = action.payload.newItem.name
      state.listTask = newListTask
    },
  },
})

export const { addTask, editTask } = TaskSlice.actions
export default TaskSlice.reducer
