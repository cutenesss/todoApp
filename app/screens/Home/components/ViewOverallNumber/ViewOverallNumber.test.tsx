import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react-native';
import { ViewOverallNumber } from '.';
import { testItemSlice } from '../../../../redux/Task/taskSlice.test';

describe("<ViewOverallNumber />", () => {
    const mockOnPressSortName = jest.fn();
    const mockOnPressSortPriority = jest.fn();
    const defaultProps = {
        totalTask: 1,
        completedTask: 1,
        onChangeSortName: mockOnPressSortName,
        onChangeSortPriority: mockOnPressSortPriority
    }

    test('renders correctly when task is not completed', () => {
        const tree = renderer.create(
            <ViewOverallNumber {...defaultProps}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('onPress works correctly', () => {
        const tree = render(<ViewOverallNumber {...defaultProps}/>);
        const itemSortName2 = tree.getByTestId('listSortName_2');
        expect(itemSortName2).toBeTruthy();
        fireEvent.press(itemSortName2);
        expect(mockOnPressSortName).toBeCalledTimes(1);
        const itemListPriorityLevel2 = tree.getByTestId('listPriorityLevel_2');
        expect(itemListPriorityLevel2).toBeTruthy();
        fireEvent.press(itemListPriorityLevel2);
        expect(mockOnPressSortPriority).toBeCalledTimes(1);
    });
});