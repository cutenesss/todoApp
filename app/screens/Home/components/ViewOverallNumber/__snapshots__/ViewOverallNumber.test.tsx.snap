// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewOverallNumber /> renders correctly when task is not completed 1`] = `
<View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 10,
        "paddingHorizontal": 20,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 19,
        }
      }
    >
      Total tasks: 1
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 19,
        }
      }
    >
      Completed tasks: 1
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 10,
        "paddingHorizontal": 20,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "width": "50%",
        }
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 4,
            "marginTop": 16,
          }
        }
      >
        Sort by priority
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
            },
            Object {
              "flexDirection": "column",
            },
            Object {
              "alignItems": "flex-start",
            },
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "width": "50%",
        }
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 4,
            "marginTop": 16,
          }
        }
      >
        Sort by name
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
            },
            Object {
              "flexDirection": "column",
            },
            Object {
              "alignItems": "flex-start",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
